p2p/session.go

```diff
@@ -286,23 +286,24 @@ func (s *session[H]) verify(headers []H) error {
 	trusted := s.from
 	// verify that the whole range is valid and adjacent.
 	for _, untrusted := range headers {
-		err := trusted.Verify(untrusted)
-		if err != nil {
-			return err
-		}
-
 		// extra check for the adjacency should be performed only for the received range,
-		// because headers are received out of order and `s.from` can't be adjacent to them
+		// because headers are received out of order and `s.from` can't be adjacent to them.
 		if trusted.Height() != s.from.Height() {
 			if trusted.Height()+1 != untrusted.Height() {
-				// Exchange requires requested ranges to always consist of adjacent headers
-				return fmt.Errorf("peer sent valid but non-adjacent header. expected:%d, received:%d",
+				// Exchange requires requested ranges to always consist of adjacent headers.
+				return fmt.Errorf("peer sent valid??? but non-adjacent header. expected:%d, received:%d",
```

Comment:
Not sure should this be `verified` or indeed `valid` (afair we validate somewhere above on the call stack).

However, after moving verification below (`err := trusted.Verify(untrusted)`) this can be only `valid`.

Reply:
Yep, they're validated in `processResponses`.